src/display/editor/alt_text.js

```diff
@@ -87,6 +109,35 @@ class AltText {
     return !this.#altText && !this.#altTextDecorative;
   }
 
+  toggleAltTextBadge(visibility) {
+    if (!this.#useNewAltTextFlow || this.#altText) {
+      if (this.#badge) {
+        this.#badge.remove();
+        this.#badge = null;
+      }
+      return;
+    }
+    if (!this.#badge) {
+      const badge = (this.#badge = document.createElement("div"));
+      badge.className = "noAltTextBadge";
+      this.#editor.div.append(badge);
+    }
+    this.#badge.classList.toggle("hidden", !visibility);
+  }
+
+  serialize(isForCopying) {
+    let altText = this.#altText;
+    if (!isForCopying && this.guessedText === altText) {
+      altText = this.#aiPrefix.replace("{ $generatedAltText }", altText);
```

Comment:
This doesn't feel very safe. Can you use `data-l10n-attrs` to pass the variable?

Reply:
I did that because the generated string (from l10n engine) was containing some markers which are not in the ascii range.
So instead of workarounding, I fixed the bug:
https://github.com/mozilla/pdf.js/pull/18493